docker/tarfile/src.go

```diff
@@ -196,7 +196,7 @@ func findTarComponent(inputFile io.Reader, path string) (*tar.Reader, *tar.Heade
 		if err != nil {
 			return nil, nil, err
 		}
-		if h.Name == path {
+		if path.Clean(h.Name) == path.Clean(rpath) {
```

Comment:
path.Clean(rpath)  should only be done once.  Not every time though the loop

Reply:
I've tried to pull this out of the loop